### PR TITLE
trueseeing: fix build

### LINKED ISSUE #281164: Build failure: trueseeing
### Steps To Reproduce

1. build trueseeing
2. Runtime dependency check fails due to missing termcolor.

### Build log

```bash
copying path '/nix/store/w2lcxb320jjziy7h0j0b0fy626hbln4x-samba-4.19.4' from 'https://cache.nixos.org'...
* Getting build dependencies for wheel...
  - termcolor not installed
  - urllib3<2.0.0,>=1.25.8 not satisfied by version 2.1.0
  - websockets<11.0,>=10.0 not satisfied by version 12.0
  - pyfiglet<0.9,>=0.8.post1 not satisfied by version 1.0.2
error: builder for '/nix/store/y2wv3y1f441m2imsh1ji0c3r9inq54fi-trueseeing-2.1.9.drv' failed with exit code 1;
       last 10 log lines:
       > Creating a wheel...
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built trueseeing-2.1.9-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for trueseeing-2.1.9-py3-none-any.whl
       >   - termcolor not installed
       For full logs, run 'nix log /nix/store/y2wv3y1f441m2imsh1ji0c3r9inq54fi-trueseeing-2.1.9.drv'.
```

Full logs : 
<summary> 
<details> 
<pre>
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/p05bg72wb6ihkj9jjmdgzsnjifd0m0ar-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/trueseeing/signature/security.py
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
* Getting build dependencies for wheel...
* Building wheel...
Successfully built trueseeing-2.1.9-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for trueseeing-2.1.9-py3-none-any.whl
  - termcolor not installed
</pre>
</details>
</summary>

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.71-1-lts, Garuda Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.1`
 - channels(root): `""`
 - channels(user): `"home-manager, nixpkgs"`
 - nixpkgs: `/home/user/.nix-defexpr/channels/nixpkgs`
---

I believe trueseeing is missing the termcolor dependency hence the failure happening during the build. @fabaff 

I can try and PR a fix but I did not try and reproduce the bug under NixOS.

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
fixes https://github.com/NixOS/nixpkgs/issues/281164

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/security/trueseeing/default.nix [hunk 1]
```diff
@@ -15,22 +15,20 @@ python3.pkgs.buildPythonApplication rec {
...
   };
 
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace "attrs~=21.4" "attrs>=21.4"
-  '';
-
   nativeBuildInputs = with python3.pkgs; [
     flit-core
+    pythonRelaxDepsHook
```
Comment (AkechiShiro) [inline pkgs/tools/security/trueseeing/default.nix:20]: Why was this added ? I believe it can be used when we want to be lax on runtime dependencies ?

File: pkgs/tools/security/trueseeing/default.nix [hunk 2]
```diff
@@ -15,22 +15,20 @@ python3.pkgs.buildPythonApplication rec {
...
+  pythonRelaxDeps = true;
+
   propagatedBuildInputs = with python3.pkgs; [
     attrs
-    ipython
     jinja2
     lxml
     pypubsub
     pyyaml
+    termcolor
```
Comment (AkechiShiro) [inline pkgs/tools/security/trueseeing/default.nix:31]: I don't understand why `pythonRelaxDeps` is used since `termcolor` is added, and the failure of the build initially was due to missing `termcolor`.

Comment (dotlambda) [reply pkgs/tools/security/trueseeing/default.nix:31]: Their pyproject.toml specifies an outdated version of termcolor.

Comment (AkechiShiro) [reply pkgs/tools/security/trueseeing/default.nix:31]: So even by adding termcolor as a dependency, it fails to build thus is `pythonRelaxDeps` necessary ?

Comment (AkechiShiro) [reply pkgs/tools/security/trueseeing/default.nix:31]: We should probably open an issue upstream to let them know that they should bump the termcolor dependency, could you take care of it ?

Comment (AkechiShiro) [reply pkgs/tools/security/trueseeing/default.nix:31]: @dotlambda Bump was merged upstream, can you please upgrade your PR ? to remove the pythonRelaxDeps ?

Comment (dotlambda) [reply pkgs/tools/security/trueseeing/default.nix:31]: There hasn't been a new release and it would break again once termcolor is updated to 2.5.

Comment (AkechiShiro) [reply pkgs/tools/security/trueseeing/default.nix:31]: I will ask for a new release then, if that's possible, if not then, we'll leave it as that.

Comment (dotlambda) [reply pkgs/tools/security/trueseeing/default.nix:31]: I will merge this anyway because, as I said, it would break again in the future.

Comment (AkechiShiro) [reply pkgs/tools/security/trueseeing/default.nix:31]: Here is the new release : https://github.com/alterakey/trueseeing/pull/358#issuecomment-1900368565 @dotlambda
